server/server.js

```diff
@@ -41,6 +43,23 @@ if (process.env.ENV === "PRODUCTION") {
   });
 }
 
+app.get("/authenticate/user-posts", (req, res) => {
+  const { ghUsername, githubId, avatarUrl } = req.query;
+  User.find({ username: ghUsername })
+    .then(async (result) => {
+      if (result.length == 0) {
+        const newUser = new User({ githubId, username: ghUsername, avatarUrl });
+        const savedUser = await newUser.save();
+      }
```

Comment:
I think it would be better to instantiate the user in `/authenticate`, right after getting a response from github. Specifically in this code snippet 
```javascript
    .then((response) => {
      return res.status(200).json(response);
    })
```

With the current code, the flow is as follows: 
1. login via Github
2. return user data given by github
3. use the github user data to fetch posts
4. create User in DB if needed while fetching posts

I think a better flow would be:
1. login via github
2. while still in the backend, use the user data from github to either:
   1. create new User in DB if not already there
   2. find the User from the DB
3. regardless of (i) or (ii), return the user object **from the DB** 
3. fetch all posts using returned user data

The idea behind the latter is to keep all user logic in `/authenticate`, including adding to the DB if necessary. Another reason is that we actually don't need a lot of the info returned by github. What we do need is everything in the User schema, so it can be easily accomplished if we just return the User object from the DB. (Note: we're actually missing the user's `pinnedPosts` with the current approach because the DB stores this, not github)